Code/Framework/AzNetworking/AzNetworking/Utilities/EncryptionCommon.cpp

```diff
@@ -100,19 +100,35 @@ namespace AzNetworking
 
     static void GetCertificatePaths(TrustZone trustZone, AZStd::string& certificatePath, AZStd::string& privateKeyPath)
     {
-        const AZ::CVarFixedString certificateFile = (trustZone == TrustZone::ExternalClientToServer) ? net_SslExternalCertificateFile : net_SslInternalCertificateFile;
-        const AZ::CVarFixedString privateKeyFile = (trustZone == TrustZone::ExternalClientToServer) ? net_SslExternalPrivateKeyFile : net_SslInternalPrivateKeyFile;
-
-        AZStd::string assetDir;
-        if (AZ::IO::FileIOBase::GetInstance() != nullptr)
+        if (auto console = AZ::Interface<AZ::IConsole>::Get(); console != nullptr)
         {
-            char buffer[AZ_MAX_PATH_LEN];
-            AZ::IO::FileIOBase::GetInstance()->ResolvePath("@products@/", buffer, sizeof(buffer));
-            assetDir = AZStd::string(buffer);
-        }
+            AZ::CVarFixedString certificateFile = "";
+            AZ::CVarFixedString privateKeyFile = "";
+            if (trustZone == TrustZone::ExternalClientToServer)
+            {
+                console->GetCvarValue("net_SslExternalCertificateFile", certificateFile);
+                console->GetCvarValue("net_SslExternalPrivateKeyFile", privateKeyFile);
+            }
+            else
+            {
+                console->GetCvarValue("net_SslInternalCertificateFile", certificateFile);
+                console->GetCvarValue("net_SslInternalPrivateKeyFile", privateKeyFile);
+            }
 
-        certificatePath = assetDir + certificateFile.c_str();
-        privateKeyPath = assetDir + privateKeyFile.c_str();
+            // Check asset directory when provided file paths are not valid
+            AZStd::string assetDir = "";
+            if (!AZ::IO::SystemFile::Exists(certificateFile.c_str()) &&
+                !AZ::IO::SystemFile::Exists(privateKeyFile.c_str()) &&
+                AZ::IO::FileIOBase::GetInstance() != nullptr)
+            {
+                char buffer[AZ_MAX_PATH_LEN];
+                AZ::IO::FileIOBase::GetInstance()->ResolvePath("@products@/", buffer, sizeof(buffer));
+                assetDir = AZStd::string(buffer);
+            }
+
+            certificatePath = assetDir + certificateFile.c_str();
```

Comment:
nit: Can you assume safe path building here? Why not use standard filepath joining as this code assume assetDir is always terminated correctly.

Reply:
this part is the original behavior, what I change is to check cvar first and if cvar value is a valid file path, then we don't use asset directory.
for whether joining is a correct way, I just keep it consistent as before to make sure it doesn't break any existing test